### ISSUE TITLE
Tilde syntax should not expand inside quoted code

### ISSUE THREAD

**Issue by alex-lew** [probcomp/Gen.jl#235]
If a Gen model includes a tilde expression in a quoted expression, then it is incorrectly replaced with `@trace`:

```julia
@gen function f()
   expr = :(x ~ normal(0, 1))
   return expr
end

f()
# => :(x = @trace(normal(0, 1), :x))

# Output should be: :(x ~ normal(0, 1))
```

I didn't realize `MacroTools.postwalk` would send us the insides of quoted expressions as if they weren't quoted! (Or maybe the problem is `MacroTools.@capture` being overly broad, and matching quoted expressions?)

This suggests that other places we use `MacroTools.postwalk` (e.g., in the `@dist` DSL) may have the same bug.

**Comment by ztangent** [comment]
I think `postwalk` descends into quotes, yes! So yeah, we have to be careful -- I think the static modelling language will have these issues as well. I think we could try and fix this together with #222 in PR #223.

**Comment by ztangent** [comment]
Addressed by #223.
